### PR TITLE
Fix node v18+ .nprmc private registry E401 issue

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Resolves: https://github.com/jfrog/jfrog-cli/issues/1844
Fixes this issue:
image

This bug occurs since Node v18 issues, declaring private registries must be scoped with the prefix: `//my-custom-registry.org/unique/path:`
More info: https://docs.npmjs.com/cli/v9/configuring-npm/npmrc?v=true#auth-related-configuration

### CODE REVIEW

File: artifactory/commands/npm/common.go [hunk 1]
```diff
@@ -155,6 +147,42 @@ func (com *CommonArgs) prepareConfigData(data []byte) ([]byte, error) {
 	return []byte(strings.Join(filteredConf, "")), nil
 }
 
+func (com *CommonArgs) processConfigLine(configLine string) (filteredLine string, err error) {
```
Comment (yahavi) [inline artifactory/commands/npm/common.go:150]:
```suggestion
func (ca *CommonArgs) processConfigLine(configLine string) (filteredLine string, err error) {
```

File: artifactory/commands/npm/common.go [hunk 2]
```diff
@@ -155,6 +147,42 @@ func (com *CommonArgs) prepareConfigData(data []byte) ([]byte, error) {
...
+
+	return fmt.Sprintf("%s\n", configLine), err
+}
+
+func (com *CommonArgs) setNpmConfigAuthEnv(value string) error {
+	// Check if the npm version is bigger or equal to 9.3.1
+	if com.npmVersion.Compare(npmVersionForLegacyEnv) <= 0 {
+		// Get registry name without the protocol name but including the '//'
+		registryWithoutProtocolName := com.registry[strings.Index(com.registry, "://")+1:]
+		// Set "npm_config_//<registry-url>:_auth" environment variable to allow authentication with Artifactory when running postinstall scripts on subdirectories.
```
Comment (yahavi) [inline artifactory/commands/npm/common.go:177]: I'm not sure this comment is true

File: artifactory/commands/npm/installorci_test.go [hunk 3]
```diff
@@ -54,6 +56,43 @@ func TestPrepareConfigData(t *testing.T) {
...
+	testCases := []struct {
+		name        string
+		com         *CommonArgs
+		value       string
+		expectedEnv string
+	}{
+		{
+			name: "set scoped registry auth env",
+			com: &CommonArgs{
+				npmVersion: version.NewVersion("9.5.0"),
```
Comment (yahavi) [inline artifactory/commands/npm/installorci_test.go:73]: It is better to test the edge cases
```suggestion
				npmVersion: version.NewVersion("9.3.1"),
```

File: artifactory/commands/npm/installorci_test.go [hunk 4]
```diff
@@ -54,6 +56,43 @@ func TestPrepareConfigData(t *testing.T) {
...
+			expectedEnv: "npm_config__auth",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.com.setNpmConfigAuthEnv(tc.value)
+			assert.NoError(t, err)
+			envValue := os.Getenv(tc.expectedEnv)
+			assert.Equal(t, tc.value, envValue)
```
Comment (yahavi) [inline artifactory/commands/npm/installorci_test.go:95]: Let's add a cleanup
```suggestion
			assert.Equal(t, tc.value, envValue)
			assert.NoError(t, os.Unsetenv(tc.expectedEnv))
```

File: go.mod [hunk 5]
```diff
@@ -94,6 +94,4 @@ require (
 
 // replace github.com/jfrog/jfrog-client-go => github.com/jfrog/jfrog-client-go v1.26.3-0.20230220094255-b9b7860406b3
 
-// replace github.com/jfrog/build-info-go => github.com/jfrog/build-info-go v1.8.5-0.20230103131235-4993ad739dc6
-
```
Comment (yahavi) [inline go.mod:98]: Let's keep this comment. It is useful.

**Review by yahavi — CHANGES_REQUESTED** [review]

File: artifactory/commands/npm/common.go (hunk 1 above)
Comment (omerzi) [reply artifactory/commands/npm/common.go:150]: done

File: artifactory/commands/npm/common.go (hunk 2 above)
Comment (omerzi) [reply artifactory/commands/npm/common.go:177]: done

File: artifactory/commands/npm/installorci_test.go (hunk 4 above)
Comment (omerzi) [reply artifactory/commands/npm/installorci_test.go:95]: done

File: artifactory/commands/npm/installorci_test.go (hunk 3 above)
Comment (omerzi) [reply artifactory/commands/npm/installorci_test.go:73]: done
